### PR TITLE
[GH-300]: Fixed the JS error while posting meeting link to thread view

### LINKED ISSUE #300: Zoom plugin while viewing a thread in the Threads view causes a JS error
Steps to reproduce:

1. Have CRT enabled, and open the Threads view
2. Open a thread that pops into the RHS
3. Click the Zoom button on the right

Expected: Zoom meeting should post into the thread
Observed: Zoom meeting isn’t posted and a JS error occurs

```
Uncaught TypeError: Cannot read properties of undefined (reading 'id')
    at Object.action (zoom_249d6071dc514b5a_bundle.js:2:360339)
    at onClick (app_bar_plugin_component.tsx:113:21)
    at Object.Be (react-dom.production.min.js:52:317)
    at Ke (react-dom.production.min.js:52:471)
    at react-dom.production.min.js:53:35
    at Pr (react-dom.production.min.js:100:68)
    at Nr (react-dom.production.min.js:101:380)
    at react-dom.production.min.js:113:65
    at De (react-dom.production.min.js:292:189)
    at react-dom.production.min.js:50:57
```

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
#### Summary
Fixed the JS error which was caused while viewing a thread in the Threads view of the zoom plugin.

#### What to test?
- Whether we are able to create the Zoom meeting in the Threads view with CRT enabled without any errors.
- Whether we are able to create the Zoom meeting in the Channel view without any errors. 

###### Steps to reproduce
1. Have CRT enabled, and open the Threads view
2. Open a thread that pops into the RHS
3. Click the Zoom button on the right

###### Environment:
MM version: v7.8.2
Node version: 14.18.0
Go version: 1.19.0

#### Ticket Link
Fixes #300

### CODE REVIEW

File: webapp/src/client/client.js [hunk 1]
```diff
@@ -11,15 +11,20 @@ export default class Client {
...
     }
 
-    forceStartMeeting = async (channelId, personal = true, topic = '', meetingId = 0) => {
-        const meetingUrl = await this.startMeeting(channelId, personal, topic, meetingId, true);
+    forceStartMeeting = async (channelId, rootId, personal = true, topic = '', meetingId = 0) => {
+        const meetingUrl = await this.startMeeting(channelId, rootId, personal, topic, meetingId, true);
         return meetingUrl;
     }
+
+    getChannelId = async (baseURL, threadId) => {
```
Comment (mickmister) [inline webapp/src/client/client.js:24]:
```suggestion
    getChannelIdForThread = async (baseURL, threadId) => {
```

File: webapp/src/index.js [hunk 2]
```diff
@@ -27,8 +27,16 @@ class Plugin {
...
                 iconURL,
-                (channel) => {
-                    startMeeting(channel.id)(store.dispatch, store.getState);
+                async (channel) => {
+                    if (channel) {
+                        startMeeting(channel.id, '')(store.dispatch, store.getState);
+                    } else {
+                        const teamId = store.getState().entities.teams.currentTeamId;
+                        const threadId = store.getState().views.threads.selectedThreadIdInTeam[teamId];
+                        const baseURL = store.getState().entities.general.config.SiteURL;
```
Comment (mickmister) [inline webapp/src/index.js:36]: I would prefer these to be in a separate file for selectors, but it's fine to have them here like this. Just one suggestion to set the state to a local var

```suggestion
                        const state = store.getState();
                        const teamId = state.entities.teams.currentTeamId;
                        const threadId = state.views.threads.selectedThreadIdInTeam[teamId];
                        const baseURL = state.entities.general.config.SiteURL;
```

**Review by mickmister — APPROVED** [review]
Great work on this @Kshitij-Katiyar :+1:

LGTM, assuming `getChannelIdForThread` concern is addressed

**Review by AayushChaudhary0001 — APPROVED** [review]
Tested and approved
Working fine, LGTM!

**Pull request merged**